test/e2e/snaps/test-snap-siginsights.spec.js

```diff
@@ -11,494 +12,723 @@ const FixtureBuilder = require('../fixture-builder');
+            tag: 'span',
+          });
+          await driver.clickElement('.mm-checkbox__input-wrapper');
+
+          // click sign button and wait for window to close
+          await driver.clickElementAndWaitForWindowToClose(
+            '[data-testid="snapInsightsButtonConfirm"]',
+          );
+
+          // switch back to test-dapp window
+          await driver.switchToWindowWithTitle(WINDOW_TITLES.TestDApp);
+
+          // check result of test
+          await driver.waitForSelector({
+            text: '0xa10b6707dd79e2f1f91ba243ab7abe15a46f58b052ad9cec170c5366ef5667c447a87eba2c0a9d4c9fbfa0a23e9db1fb55865d0568c32bd7cc681b8d0860e7af1b',
+            tag: 'span',
+          });
+
+          // TEST TWO: sign typed data
+          // find and scroll to sign typed data and click sign
+          const signTypedButton1 = await driver.findElement('#signTypedData');
+          await driver.scrollToElement(signTypedButton1);
+          await driver.clickElement('#signTypedData');
+
+          // switch back to MetaMask window and switch to tx insights pane
+          await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
+
+          // wait for and click sign
+          await clickSignOnSignatureConfirmation({
+            driver,
+            snapSigInsights: true,
+          });
+
+          // look for returned signature insights data
+          await driver.waitForSelector({
+            text: '1',
+            tag: 'p',
+          });
+
+          // wait for host to render and click checkbox to authorize signing
+          await driver.waitForSelector({
+            text: '127.0.0.1:8080',
+            tag: 'span',
+          });
+          await driver.clickElement('.mm-checkbox__input-wrapper');
+
+          // click sign button and wait for window to close
+          await driver.clickElementAndWaitForWindowToClose(
+            '[data-testid="snapInsightsButtonConfirm"]',
+          );
+
+          // switch back to test-dapp window
+          await driver.switchToWindowWithTitle(WINDOW_TITLES.TestDApp);
+
+          // check result of test
+          await driver.waitForSelector({
+            text: '0x32791e3c41d40dd5bbfb42e66cf80ca354b0869ae503ad61cd19ba68e11d4f0d2e42a5835b0bfd633596b6a7834ef7d36033633a2479dacfdb96bda360d51f451b',
+            tag: 'span',
+          });
+
+          // TEST THREE: sign typed data v3
+          // find and scroll to sign typed data v3 and click sign
+          const signTypedV3Button1 = await driver.findElement(
+            '#signTypedDataV3',
+          );
+          await driver.scrollToElement(signTypedV3Button1);
+          await driver.clickElement('#signTypedDataV3');
+
+          // switch back to MetaMask window and switch to tx insights pane
+          await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
+
+          // click down arrow
+          await driver.waitForSelector('.fa-arrow-down');
+          await driver.clickElement('.fa-arrow-down');
```

Comment:
Are we guaranteed that the scroll is required each time? On the new tests we use `clickElementSafe` in case of certain edge-cases where scrolling is not required for whatever reason. We had some flakiness in the past without doing this